semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/span/SpanStore.java

```diff
@@ -0,0 +1,118 @@
+  private static final String SPAN_STORE_REDIS_USER = "span.store.redis.user";
+  private static final String SPAN_STORE_REDIS_PASSWORD = "span.store.redis.password";
+
+  private static final ByteBuffer TRACE_ID = ByteBuffer.wrap("t".getBytes());
+
+  private static final String IS_TRIM = "is_trim";
+  private static final AttributeValue ATTRIBUTE_VALUE_FALSE =
+      AttributeValue.newBuilder().setValue("false").build();
+  private static final AttributeValue ATTRIBUTE_VALUE_TRUE =
+      AttributeValue.newBuilder().setValue("true").build();
+  private static final String SPAN_BYPASSED_CONFIG = "processor.bypass.key";
+  private final Optional<Jedis> jedis;
+  private Optional<String> bypassKey;
+
+  public SpanStore(Config config) {
+    boolean enabled =
+        config.hasPath(SPAN_STORE_ENABLED) ? config.getBoolean(SPAN_STORE_ENABLED) : false;
+    if (enabled) {
+      DefaultJedisClientConfig defaultJedisClientConfig =
+          DefaultJedisClientConfig.builder()
+              .password(config.getString(SPAN_STORE_REDIS_PASSWORD))
+              .build();
+      jedis =
+          Optional.of(
+              new Jedis(
+                  config.getString(SPAN_STORE_REDIS_HOST),
+                  config.getInt(SPAN_STORE_REDIS_PORT),
+                  defaultJedisClientConfig));
+      bypassKey =
+          config.hasPath(SPAN_BYPASSED_CONFIG)
+              ? Optional.of(config.getString(SPAN_BYPASSED_CONFIG))
+              : Optional.empty();
+    } else {
+      jedis = Optional.empty();
+    }
+  }
+
+  public Event pushToSpanStoreAndTrimSpan(Event event) {
+    try {
+      if (jedis.isPresent()) {
+        SpanIdentity spanIdentity =
+            SpanIdentity.newBuilder()
+                .setTenantId(event.getCustomerId())
+                .setTraceId(TRACE_ID)
+                .setSpanId(event.getEventId())
+                .build();
+        Event eventCopy = Event.newBuilder(event).build();
+        jedis.get().set(spanIdentity.toByteBuffer().array(), eventCopy.toByteBuffer().array());
+        Optional<AttributeValue> attributeValue =
+            bypassKey.flatMap(
+                key -> Optional.ofNullable(event.getAttributes().getAttributeMap().get(key)));
+        Map<String, AttributeValue> attributesMap =
+            attributeValue
+                .map(value -> Map.of(bypassKey.get(), value))
+                .orElse(Collections.emptyMap());
+        return Event.newBuilder(event)
+            .setAttributes(Attributes.newBuilder().setAttributeMap(attributesMap).build())
+            .setEnrichedAttributes(
+                Attributes.newBuilder()
+                    .setAttributeMap(Map.of(IS_TRIM, ATTRIBUTE_VALUE_TRUE))
+                    .build())
+            .build();
+      }
+    } catch (IOException e) {
+      LOGGER.error("Unable to push event to span store", e);
+    }
+
+    return event;
+  }
+
+  public Event retrieveFromSpanStoreAndFillSpan(Event event) {
```

Comment:
Fetching each span from store would involve lot of network IO. We may need batch gets in order to scale. https://redis.io/commands/mget/